app/src/main/java/org/moire/opensudoku/gui/inputmethod/IMControlPanel.java

```diff
@@ -215,9 +214,9 @@ private void ensureInputMethods() {
 
 	private void createInputMethods() {
 		if (mInputMethods.size() == 0) {
-			addInputMethod(INPUT_METHOD_POPUP, new IMPopup());
-			addInputMethod(INPUT_METHOD_SINGLE_NUMBER, new IMSingleNumber());
-			addInputMethod(INPUT_METHOD_NUMPAD, new IMNumpad());
+			addInputMethod(INPUT_METHOD_POPUP.ordinal(), new IMPopup());
+			addInputMethod(INPUT_METHOD_SINGLE_NUMBER.ordinal(), new IMSingleNumber());
+			addInputMethod(INPUT_METHOD_NUMPAD.ordinal(), new IMNumpad());
```

Comment:
Use of ordinals can cause unintended side effects when re-arranging Enums. You should extend the enum to have assigned values, then it is obvious that the enums are ordered and this issue can be noticed more easily.